### PR TITLE
Vertical products

### PR DESCRIPTION
**Pull request by hejohns**

### CODE REVIEW

File: Cubical/Categories/Limits/BinProduct/More.agda [hunk 1]
```diff
@@ -205,6 +240,12 @@ module _ (C : Category ℓ ℓ') where
       ×pF-with-agrees : ×Bif ⟪ C .id , f ⟫× ≡ PWN.×pF ⟪ f ⟫
       ×pF-with-agrees = sym (×Bif .Bif-R×-agree _)
 
+  module BinProducts'Notation (bp : BinProducts') =
+    Notation (BinProducts'ToBinProducts bp)
+
+-- TODO: define Notation in terms of BinProduct c c',
```
Comment (hejohns) [inline Cubical/Categories/Limits/BinProduct/More.agda:246]: I'm thinking about doing this

File: Cubical/Categories/Limits/BinProduct/More.agda [hunk 2]
```diff
@@ -140,25 +174,63 @@ module _ (C : Category ℓ ℓ') where
       ×-extensionality : π₁ ∘⟨ C ⟩ f ≡
                          π₁ ∘⟨ C ⟩ g → π₂ ∘⟨ C ⟩ f ≡ π₂ ∘⟨ C ⟩ g → f ≡ g
       ×-extensionality p1 p2 = extensionality (ues _) (ΣPathP (p1 , p2))
-  module Notation (bp : BinProducts C) where
+
+  module NotationAt {a b : C .ob} (bp : BinProduct C a b) where
```
Comment (hejohns) [inline Cubical/Categories/Limits/BinProduct/More.agda:178]: Moved much of Notation into NotationAt as possible, except for things that require all products like the bifunctoriality of (- × -) : C × C → C

**Comment by hejohns** [comment]
After one more cleaning pass, this is ready for review

File: Cubical/Categories/Limits/BinProduct/More.agda (hunk 1 above)
Comment (maxsnew) [reply Cubical/Categories/Limits/BinProduct/More.agda]: did you?

File: Cubical/Categories/Displayed/Fibration/Properties.agda [hunk 3]
```diff
@@ -1,33 +1,2 @@
 {-# OPTIONS --safe #-}
 module Cubical.Categories.Displayed.Fibration.Properties where
```
Comment (maxsnew) [inline Cubical/Categories/Displayed/Fibration/Properties.agda:2]: should this file be deleted?

File: Gluing/CartesianCategory.agda [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+open import Cubical.Foundations.Prelude
+open import Cubical.Relation.Nullary hiding (⟪_⟫)
+open import Cubical.Data.Nat
+open import Cubical.Data.Nat.Properties
+open import Cubical.Data.Bool
+open import Cubical.Data.Sum as Sum
+
+open import Cubical.Categories.Category renaming (isIso to isIsoC)
+open import Cubical.Categories.Functor
+open import Cubical.Categories.Constructions.Free.CartesianCategory.Base as Law
```
Comment (maxsnew) [inline Gluing/CartesianCategory.agda:14]: Law?

**Comment by maxsnew** [comment]
I just tweaked the gluing example slightly to make the generator a single morphism from 1 to ans x ans, so that it uses the products in a slightly non-trivial way

File: Cubical/Categories/Displayed/Fibration/Properties.agda (hunk 3 above)
Comment (hejohns) [reply Cubical/Categories/Displayed/Fibration/Properties.agda]: now removed by 27beb225e9c74654820d60dae934796e05dc8ccf

**Comment by hejohns** [comment]
> I just tweaked the gluing example slightly to make the generator a single morphism from 1 to ans x ans, so that it uses the products in a slightly non-trivial way

Great, thanks for doing that

File: Cubical/Categories/Limits/BinProduct/More.agda (hunk 1 above)
Comment (hejohns) [reply Cubical/Categories/Limits/BinProduct/More.agda]: yes, remarked at https://github.com/maxsnew/cubical-categorical-logic/pull/94#discussion_r1651574537

File: Gluing/CartesianCategory.agda (hunk 4 above)
Comment (hejohns) [reply Gluing/CartesianCategory.agda:14]: Short for Lawvere, as in Lawvere theory or Category with Finite Products

File: Cubical/Categories/Displayed/Limits/BinProduct.agda [hunk 5]
```diff
@@ -11,25 +13,305 @@ open import Cubical.Categories.Displayed.Base
...
+          fᴰ* : ∃![ fᴰ* ∈ Cᴰ.Hom[ h ][ θ , B*.ϕ[π₁x] ] ] fᴰ* Cᴰ.⋆ᴰ B*.π₁* ≡ fᴰ
+          fᴰ* = B*.fᴰ* fᴰ
+
+          gᴰ* : ∃![ gᴰ* ∈ Cᴰ.Hom[ h ][ θ , B*.ψ[π₂x] ] ] gᴰ* Cᴰ.⋆ᴰ B*.π₂* ≡ gᴰ
+          gᴰ* = B*.gᴰ* gᴰ
+
+          hᴰ : Cᴰ.Hom[ h ][ θ , ϕ[π₁x]∧ψ[π₂x].vert ]
+          hᴰ = ϕ[π₁x]∧ψ[π₂x].⟨ fᴰ* .fst .fst , gᴰ* .fst .fst ⟩'
+
+          goal : ∀ hᴰ' p → hᴰ ≡
```
Comment (maxsnew) [inline Cubical/Categories/Displayed/Limits/BinProduct.agda:251]: Any thoughts on how to make this proof less nasty?

**Comment by maxsnew** [comment]
whoops meant to squash and merge, oh well.

**Pull request merged**